src/core/compact_object.cc

```diff
@@ -1135,15 +1137,17 @@ void CompactObj::GetString(char* dest) const {
     } else if (taglen_ == SMALL_TAG) {
       size_t decoded_len = DecodedLen(u_.small_str.size());
 
+      // we left some space on the left to allow inplace ascii unpacking.
+      size_t space_left = decoded_len - u_.small_str.size();
+
       string_view slices[2];
       unsigned num = u_.small_str.GetV(slices);
       DCHECK_EQ(2u, num);
-      std::string tmp(decoded_len, ' ');
-      char* next = tmp.data();
+      char* next = dest + space_left;
```

Comment:
`taglen_ == SMALL_TAG` so the string max len for that path after encoding is 255 bytes. Without encoding and assuming a 20% reduction, the maximum for the `tmp` buffer would have been `320 bytes` since a 20% reduction on that would give us `256` bytes and `SmallString::CanAllocate` limit is `256` bytes.

The 20% is also too much. bitpack is roughly 12.5% based on your comment and for huffman 1.8 bits per byte (haven't any serious research on this so it might be a little off)

Reply:
Yes, we are lucky that it's only hundreds of bytes